### PR TITLE
JIRA# ROS-261 - Scripted Smart Commits

### PR DESCRIPTION
**Pull request by DoSOfRedRiver**
Changed the "workflow" of the script, now it should be more intuitive and less verbose.
Changed the code structure. I gave up of self-made cmd parsing in favor of `argparse` module.  Also, I created a little DSL for `argparse` to make it's configuration clearer.

### CODE REVIEW

**Comment by kirkwood** [comment]
Big +1 for using `argparse`.

File: scripts/pr/pr.py [hunk 1]
```diff
@@ -260,16 +284,220 @@ def executeTask(self, args):
...
+
+
+class Workflow:
+
+    def __init__(self, config, tasks):
+        self.config = config
+        self.tasks = tasks
+        self.commitMsg = {
+            'comment': self.commentCommit,
+            'transite': self.transiteCommit
```
Comment (kirkwood) [inline scripts/pr/pr.py]: I might just do a global search-and-destroy for `transite` :)

Comment (DoSOfRedRiver) [reply scripts/pr/pr.py]: Oh, sure.

File: scripts/pr/pr.py [hunk 2]
```diff
@@ -260,16 +289,220 @@ def executeTask(self, args):
...
+
+
+    def storeArgs(self, args, **add):
+        d = dictDel(args.__dict__, 'func')
+        self.config.storeVariables(**dictMerge(d,add))
+
+
+    def init(self, args):
+        self.storeArgs(args, phase='init')
+        featureName = args.title.strip().replace(" ", "-").lower()
```
[the remark this answers is not in the document]
Comment (DoSOfRedRiver) [reply scripts/pr/pr.py]: Hmm, I don't see any difference with my variant.

**Comment by DoSOfRedRiver** [comment]
I will add a `--help-all` command a bit later. Thanks for the feedback!

**Comment by DoSOfRedRiver** [comment]
Many thanks, I will wait for the Nash's decision. Also, I note your recommendations for the future.

File: scripts/pr/pr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kirkwood) [reply scripts/pr/pr.py]: Ah, yes. An example!
```Python
def yours(s):
    return s.strip().replace(' ', '-').lower()

def mine(s):
    return '-'.join(s.lower().split())


print '%12s %12s %12s' % ('original', 'yours', 'mine')
print '%12s %12s %12s' % ((('-'*10), )*3)
for s in [ 'foo',
           'foo bar',
           ' foo bar ',
           'foo  bar',
           'foo\nbar',
           'foo\tbar' ]:
    print '%12r %12r %12r' % (s, yours(s), mine(s))
```
=>
```
    original        yours         mine
  ----------   ----------   ----------
       'foo'        'foo'        'foo'
   'foo bar'    'foo-bar'    'foo-bar'
 ' foo bar '    'foo-bar'    'foo-bar'
  'foo  bar'   'foo--bar'    'foo-bar'
  'foo\nbar'   'foo\nbar'    'foo-bar'
  'foo\tbar'   'foo\tbar'    'foo-bar'
```

[the remark this answers is not in the document]
Comment (DoSOfRedRiver) [reply scripts/pr/pr.py]: Got it, thanks!

**Pull request merged**